### ISSUE TITLE
Dino leaks huge amounts of memory

### ISSUE THREAD

**Issue by lorenz** [dino/dino#301]
Currently have Dino sitting at 7349MiB RSS, which indicates a rather big memory leak. Debugging GTK+ programs for leaks is hard though because GTK+ iself intentionally leaks memory. How can we track this down?

**Comment by jadahl** [comment]
Had dino running for for a few days, and noticed it had consumed ~10GB (RSS) ram. No other GTK+ application does that, so it's probably not GTK+.

**Comment by Porkepix** [comment]
Guess this would go here:

![screenshot from 2018-06-24 19-02-34](https://user-images.githubusercontent.com/709500/41821441-366658b8-77e1-11e8-8c7d-e851d85094a1.png)

This is after a bit more than 6 days uptime (on a laptop, so not 100% of the time awaken). This is quite a lot.

**Comment by andreygursky** [comment]
For valgrind friendly debugging just invoke dino like:
```
$ G_DEBUG=gc-friendly G_SLICE=always-malloc valgrind --leak-check=full dino
```
For reference: https://developer.gnome.org/glib/stable/glib-running.html

**Comment by Chobbes** [comment]
I'm having the same problem, seems to happen when I get a lot of messages about not being able to connect to the XMPP server.

**Comment by licaon-kter** [comment]
@Chobbes You've build from HEAD or OBS?

**Comment by Chobbes** [comment]
HEAD. Not sure what OBS is.

**Comment by jubalh** [comment]
> HEAD. Not sure what OBS is.

OpenBuildService. It's where we build the dino packages: https://build.opensuse.org/package/show/network:messaging:xmpp:dino/dino

**Comment by andir** [comment]
Just another report on this: I am also running into this. Using the builds included in NixOS (https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/dino/default.nix#L18). It happend with previous version as with the latest one as of a few days ago.

**Comment by Rahix** [comment]
I ran valgrind and it showed most leaks to happen somewhere inside Qlite.  Unfortunately I don't know vala well enough to tell where exactly those leaks were occuring ...

**Comment by LAGonauta** [comment]
The leak seemed to not happen on the February build, but is happening again:
```
==9211== LEAK SUMMARY:
==9211==    definitely lost: 211,739 bytes in 504 blocks
==9211==    indirectly lost: 398,934 bytes in 15,232 blocks
==9211==      possibly lost: 2,235,697 bytes in 1,398 blocks
==9211==    still reachable: 833,424,064 bytes in 297,964 blocks
==9211==                       of which reachable via heuristic:
==9211==                         length64           : 395,384 bytes in 1,326 blocks
==9211==                         newarray           : 3,008 bytes in 108 blocks
==9211==         suppressed: 0 bytes in 0 blocks
```
How to reproduce: keep clicking in any conversation on the left. It seems that conversations with more content leak faster (i have one that leaks 100MB per click)

**Comment by licaon-kter** [comment]
I can repro just by keeping CTRL-Tab pressed ;)

**Comment by fiaxh** [comment]
The comments in this issue likely referred to different memory leaks over time. For example, the issue lagonauta and licaon-kter refer to has likely been fixed in 8fe723bccb1dbcc38ec2742195cc77702f7f3f52 and there have already been memory-related fixes earlier.

I'll close this issue to make sure we only track issues that still occur: If you still have memory consumption problems with recent Dino versions, please open a new issue.
